src/unistd.rs

```diff
@@ -942,6 +942,42 @@ pub fn setgid(gid: Gid) -> Result<()> {
 }
 
 #[inline]
+pub fn setgroups(groups: &[Gid]) -> Result<()> {
+    cfg_if! {
+        if #[cfg(any(target_os = "macos",
+                     target_os = "ios",
+                     target_os = "freebsd",
+                     target_os = "dragonfly",
+                     target_os = "openbsd",
+                     target_os = "netbsd"))] {
+            type setgroups_ngroups_t = c_int;
+        } else {
+            type setgroups_ngroups_t = size_t;
+        }
+    }
+    let gids: Vec<gid_t> = groups.iter().cloned().map(Into::into).collect();
+    let res = unsafe { libc::setgroups(groups.len() as setgroups_ngroups_t, gids.as_ptr()) };
+
+    Errno::result(res).map(drop)
+}
+
+#[cfg(any(target_os = "linux", target_os = "android", target_os = "macos", target_os = "ios"))]
```

Comment:
Yes, but it's unfortunately not implemented in `rust-lang/libc` for FreeBSD yet.